server/wasabi_context.go

```diff
@@ -0,0 +1,47 @@
+package server
+
+import (
+	"context"
+	"time"
+)
+
+type WasabiContext interface {
+	context.Context
+	GetHttpConfig() HttpConfig
+}
+
+type WasabiDefaultContextImpl struct {
+	context.Context
+}
```

Comment:
Why not use std lib context? with WithValue wrapper https://pkg.go.dev/context#example-WithValue to pass values, yes in terms of performance could be not the best solution but it's closer to std lib.

IMHO not everything need to be part of context... http timeouts unlikely will be useful anywhere outside of the server, it's better to manage them inside server itself.

Reply:
yes, but we can have selectors like:

```
func GetHTTPConfig(ctx context.Context) *HTTPConfig {
   val := ctx.Value(configKey)
   if val == nil {
       return nil
   }

   config, ok := val.(*HTTPConfig)
   if !ok {
       return nil
   }

   return Config
```